CRAMv3.tex

```diff
@@ -1336,16 +1336,22 @@ \subsection{Read names (RN data series)}
 \subsection{\textbf{Mate record}}
 \label{subsec:mate}
 
-There are two ways in which mate information can be preserved in CRAM: number of records downstream (distance, within this slice) to the next fragment in the template and a special mate record if the next fragment is not in the current slice.
-In the latter case the record is labelled as ``detached'', see the CF data series.
+There are two ways in which mate information can be preserved in CRAM.
+If the next fragment is not in the same slice we store verbatim copies of the insert size, mate reference chromosome and positions, and mate flags (mapped status, orientation) for both records.
+In this case both records are labelled as ``detached'' in the CF data series using bit 2.
 
-For mates within the slice only the distance is captured, and only for the first record.  The mate has neither detached nor downstream flags set in the CF data series.
+If this and the next fragment are within the same slice, we can derive much of this information by comparing the two records.
+The upstream record has CF bit 4 (mate downstream) flag set and stores the number of records to skip (in the NF data series) between this record and the record for the next fragment on this template, with zero meaning the next fragment is also the next record.
+The downstream record has neither CF bits 2 (detached) or 4 (mate downstream) set nor does it use the NF data series (unless it also has an additional ``next fragment'' to refer to).
+
+It is not mandatory to use this deduplication approach and optionally tools may wish to label data as detached even when all records for the template reside in the same slice.
+One example reason for this may be to encode inconsistent data in order to preserve round-trips in to and out of the CRAM format.
```

Comment:
On second thought, I think I now understand the reference to inconsistent data here. Perhaps: "One reason to do this may be to preserve inconsistent data so that it round-trips though the CRAM format with full fidelity."